crates/tako/src/internal/common/ids.rs

```diff
@@ -0,0 +1,46 @@
+use crate::define_id_type;
+use serde::{Deserialize, Serialize};
+use std::fmt::{Debug, Display, Formatter};
+
+define_id_type!(JobId, u32);
+define_id_type!(JobTaskId, u32);
+define_id_type!(WorkerId, u32);
+define_id_type!(InstanceId, u32);
+
+#[derive(Default, Copy, Clone, Hash, PartialOrd, Ord, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TaskId {
+    job_id: JobId,
+    job_task_id: JobTaskId,
+}
+
+impl Display for TaskId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}@{}", self.job_id, self.job_task_id)
```

Comment:
I would use `/` as the separator. We already used it on a couple of places before, and `@` evokes "location" to me, so `X@N` could be interpreted as task X on worker N.

Reply:
It was my first implementation, but because they are number, the result looks like a fraction: "1/2", so "/" is bad idea.

Reply:
- `1-2`: looks like a range or a subtraction :)
- `[1, 2]`: looks like an array
- `1>2` ?
- `1#2`

Reply:
You are following my steps:). Actually I was reserving "#" for data objects, 1@2#3 could be a name of data object in the future. Other reasonable options are: 1`2 or 1$2, but 1@2 seems better in my eyes.

Reply:
`1$2#3@4` - Job 1, task 2, data object 3 at worker 4 :laughing: Ok, let's leave `@`.